EIPS/eip-2537.md

```diff
@@ -0,0 +1,293 @@
+---
+eip: 2537
+title: BLS12-381 curve operations
+author: Alex Vlasov (@shamatar)
+discussions-to: https://github.com/ethereum/EIPs/pull/2537
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-02-21
+requires : 1109, 2046
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+
+This precompile adds operation on BLS12-381 curve as a precompile in a set necessary to *efficiently* perform operations such as BLS signature verification and perform SNARKs verifications.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+If `block.number >= X` we introduce *eight* separate precompiles to perform the following operations (addresses to be determined) with one of them being optional and up to discussion (BLS12-381 curve allows efficient square root extraction, so point decompression makes *some* sense):
+
+- G1ADD - to perform point addition on a curve defined over prime field
```

Comment:
Should all these precompiles have a `BLS12` prefix?

In some earlier precompile ubiquitous names were not agreed on as a result different clients call them differently. Trying to avoid that here.

Reply:
Yep, added prefixes